lib/python/rose/unicode_utils.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (C) 2012-2019 British Crown (Met Office) & Contributors.
+#
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+import sys
+import io
+
+ENCODING='UTF-8'
+
+def write_safely(msg, handle):
+    """Wrap handle.write command in logic to deal with change to
+    Python3 unicode and byte types.
+
+    Args:
+        msg (str or bytes):
+            Message to be written - type uncertain.
+        handle (file handle):
+            Potentially Obscure file or file-like object. May include a variety of
+            buffer types.
+    """
+    if isinstance(handle, io.BufferedWriter) and isinstance(msg, str):
+        handle.write(msg.encode(ENCODING))
+    elif isinstance(handle, io.StringIO) and isinstance(msg, bytes):
+        handle.write(msg.decode())
+        print('N')
+    elif isinstance(handle, io.BufferedWriter) and isinstance(msg, bytes):
+        handle.write(msg)
+    elif isinstance(handle, io.StringIO) and isinstance(msg, str):
+        handle.write(msg)
+    else:
+        raise TypeError(f'Unsupported types:'
+          f'\t{msg} has type {type(msg)}'
+          f'\t{handle} has type {type(handle)}')
```

Comment:
Not a good day for polymorphism, but I don't really have any good alternate suggestion.